### PR TITLE
Add libtiff/4.1.0 recipe

### PR DESCRIPTION
**Pull request by theirix**
Specify library name and version:  **libtiff/4.1.0**

Updated to 4.1.0. Only minor changes are required.

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libtiff/all/conanfile.py [hunk 1]
```diff
@@ -43,6 +43,8 @@ def build(self):
         cmake.definitions["lzma"] = False
         cmake.definitions["jpeg"] = False
         cmake.definitions["jbig"] = False
+        cmake.definitions["webp"] = False
+        cmake.definitions["zstd"] = False
```
Comment (Croydon) [inline recipes/libtiff/all/conanfile.py]: This also disables it for 4.0.9 and may break consumers

Why this change?

Comment (theirix) [reply recipes/libtiff/all/conanfile.py]: This flag appears only in 4.1.0, therefore, it cannot affect 4.0.9.

I prefer to disable all dependencies that not specified in options because they can be accidentally picked up from system libraries on developer or CI machine.

**Comment by theirix** [comment]
Thank you!
